Consistent uEntity naming (#290)
eclipse-uprotocol/up-spec: terms.adoc

Something that can be manipulated/controlled/exposed by a service. Resources are unique when prepended with device and service.
*Static Topic:* Resource that is declared in the service proto in Resource enum and known at compile time (ex. door, window, etc...)
*Dynamic Topic:* Created at runtime and not defined in the service proto (ex. geo-fence locations)
|*uBus* |Message bus that dispatches CEs between uEntities over a common transport. It provides multicast and forwarding functionality (works like a network switch) 
|*uStreamer* |Device-2-Device event dispatcher to enable seamless communication between uEntities on different devices which might talk different transport layer protocols. Performs various additional functionality such as file transfer, event buffering and more. i.e. when events need to move form one transport to the next it flows through the streamer (can be equated to an IP router) 
|*uDiscovery* |Provides a means for uEntities to discover each other, their location (address), properties, and more.
|*Topic* |What applications subscribe to and services publish to. It is represented as a uProtocol URI containing authority, sw entity, and resource information

> |*uApplication* |A uEntity that performs the role of consumer

Reviewer: ```suggestion
|*uApplication* |A uEntity that acts as a _service consumer_
```

> |*uService* |A uEntity that performs the role of producer whos interfaces are defined with a protobuf IDL

Reviewer: ```suggestion
|*uService* |A uEntity that acts as a _service provider_. The interface describing the provided service operations is defined by means of [proto3](https://protobuf.dev/)
```
